src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt

```diff
@@ -12,18 +12,26 @@ class ReleaseDateService(
 ) {
 
   /** Friday is also considered as weekend */
-  fun isEligibleForEarlyRelease(releaseDate: LocalDate?): Boolean {
+  fun isEligibleForEarlyRelease(releaseDate: LocalDate?, includeFriday: Boolean): Boolean {
     val listOfBankHolidays: List<LocalDate> = bankHolidayService.getBankHolidaysForEnglandAndWales()
     val dayOfWeek = releaseDate?.dayOfWeek
-    if (dayOfWeek == DayOfWeek.FRIDAY || dayOfWeek == DayOfWeek.SATURDAY || dayOfWeek == DayOfWeek.SUNDAY) {
+    if (getListOfWeekends(includeFriday).any { it === dayOfWeek }) {
       return true
     }
     return listOfBankHolidays.contains(releaseDate)
   }
 
-  fun getEarliestReleaseDate(releaseDate: LocalDate) =
+  private fun getListOfWeekends(includeFriday: Boolean): MutableList<DayOfWeek> {
+    val listOfDays = mutableListOf<DayOfWeek>(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY)
+    if (includeFriday) {
+      listOfDays.add(DayOfWeek.FRIDAY)
+    }
+    return listOfDays
+  }
+
+  fun getEarliestReleaseDate(releaseDate: LocalDate, includeFriday: Boolean, days: Int? = null) =
```

Comment:
Passing [booleans](https://understandlegacycode.com/blog/what-is-wrong-with-boolean-parameters/) to change how methods work is a code smell 

think you're trying to do too many things with this method - this new feature doesn't really have anything to do with earliest release date. 

Would suggest adding a new function and leaving this as it is. 
May be a `isXWorkingDaysBefore(releaseDate, days)` ?

Reply:
you could talk to @gurps-singh about using a clock

Reply:
Ok I will add a new function and leave this as it is.